File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/core/values/Inventar.kt
package org.github.haschi.haushaltsbuch.core.values

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize

@JsonSerialize(`as` = Inventar::class)
@JsonDeserialize(`as` = Inventar::class)
@JsonIgnoreProperties(value= "reinvermoegen", allowGetters=true)
data class Inventar(
        val anlagevermoegen: Vermoegenswerte,
        val umlaufvermoegen: Vermoegenswerte,
        val schulden: Schulden)
{

    val reinvermoegen: Reinvermögen
        get() = Reinvermögen(
                Währungsbetrag(
                        anlagevermoegen.summe.wert.add(
                                umlaufvermoegen.summe.wert)),
                schulden.summe)


    companion object
    {
        val leer: Inventar
            get() = Inventar(
                    Vermoegenswerte.keine,
                    Vermoegenswerte.keine,
                    Schulden.keine)
    }
}

File: ui/frontend/src/main/src/app/inventur/inventar.ts
import { Inventarposition } from './inventarposition';

export interface Reinvermoegen {
    summeDesVermoegens: string;
    summeDerSchulden: string;
    reinvermoegen: string
}

export interface Inventar {
    anlagevermoegen: Inventarposition[];
    umlaufvermoegen: Inventarposition[];
    schulden: Inventarposition[];
    reinvermoegen: Reinvermoegen;
}
